### ISSUE TITLE
How to use the `GitbeakerRequestError` class?

### ISSUE THREAD

**Issue by domharrington** [jdalrymple/gitbeaker#3510]
**Description**

Hey! I'm trying to do some error handling to check for a given status code and ignore the error if it's a 404, but throw otherwise. My code looks something like this:

```js
import { Gitlab } from '@gitbeaker/rest';
import { GitbeakerRequestError } from '@gitbeaker/requester-utils';

const api = new Gitlab({
  host: 'https://api.mygitlab.com',
  token: '123456',
  camelize: true,
});

try {
  const res = await api.RepositoryFiles.show(1, 'readme.md', 'main');
} catch (e) {
  if (!(e instanceof GitbeakerRequestError)) throw e;
  const error = e as GitbeakerRequestError;

  if (error.cause.response.status !== 404) throw e;
}
```

I'm getting the following TS error when I try to access the `cause` property
<img width="881" alt="image" src="https://github.com/jdalrymple/gitbeaker/assets/848223/ec1266ba-3b1f-482a-a383-3b9d05759285">

I can see the class defined [here](https://github.com/jdalrymple/gitbeaker/blob/1d7e9aee42b91b36e7757c1de5fb6251f459afa8/packages/requester-utils/src/GitbeakerError.ts). Is there something else I need to do to get typing of the custom error class? 

Also it would be awesome if this class was expose on the @gitbeaker/rest package so I don't have to include another package.

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [x] I have read [the documentation](https://github.com/jdalrymple/gitbeaker/blob/1d7e9aee42b91b36e7757c1de5fb6251f459afa8/packages/rest/README.md#error-handling).

Thanks in advance!

**Comment by jdalrymple** [comment]
Hmm maybe I messed up somewhere. Ill give it a look and follow up! The way you have it written should be how it works though, so its probably an issue on my end.

**Comment by domharrington** [comment]
Thank you!! The code changes in #3513 look like they'll do the trick. One other thing i've noticed is that the `error.cause.description` seems to be double quoted:

```js
Error: Not Found
    at throwFailedRequestError (/node_modules/@gitbeaker/rest/dist/index.js:75:9)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async defaultRequestHandler (/node_modules/@gitbeaker/rest/dist/index.js:111:7)
    at async /node_modules/@gitbeaker/core/dist/index.js:111:22
  [cause]: {
    description: '"404 File Not Found"',
    request: _Request [Request] {
      [Symbol(realm)]: [Object],
      [Symbol(state)]: [Object],
      [Symbol(signal)]: [AbortSignal],
      [Symbol(abortController)]: [AbortController],
      [Symbol(headers)]: [_HeadersList]
    },
    response: _Response [Response] {
      [Symbol(realm)]: null,
      [Symbol(state)]: [Object],
      [Symbol(headers)]: [_HeadersList]
    }
  }
}
```

See the `description: '"404 File Not Found"'` is single quoting a double quoted string. Can create another issue if you'd like or happy to submit a PR!

**Comment by jdalrymple** [comment]
Yea its because im stringifying it. Ill change that real quick.

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`39.32.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/39.32.0) :rocket:

**Comment by domharrington** [comment]
Thank you! 🚀
